### PR TITLE
fix(controller): Add klog logrus bridge. Fixes #2707.

### LINKED ISSUE #2707: Logs from k8s aren't captured by logrus
Checklist:

* [ ] I've included steps to reproduce the bug.
* [ ] I've included the version of argo rollouts.

**Describe the bug**

The logging output don't include k8s logs in json format.

**To Reproduce**

Start the app with `--logformat=json`

**Expected behavior**

All logs should be formatted as json (using the same logger).

**Screenshots**

**Version**

1.4.1

**Logs**

```
# Paste the logs from the rollout controller

# Logs for the entire controller:
{"generation":320,"level":"info","msg":"Reconciliation completed","namespace":"default","resourceVersion":"5359174568","rollout":"my-app-name","time":"2023-04-10T17:16:00Z","time_ms":1723.9298860000001}
I0410 17:16:00.525100       1 trace.go:205] Trace[268113179]: "DeltaFIFO Pop Process" ID:default/another-app-here,Depth:424,Reason:slow event handlers blocking the queue (10-Apr-2023 17:16:00.349) (total time: 175ms):
Trace[268113179]: [175.828219ms] [175.828219ms] END
{"level":"info","msg":"Start processing","resource":"default/my-ui-name","time":"2023-04-10T17:16:00Z"}

# Logs for a specific rollout:
kubectl logs -n argo-rollouts deployment/argo-rollouts | grep rollout=<ROLLOUTNAME
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

### PR DESCRIPTION
**Pull request by Sineaggi**
closes: https://github.com/argoproj/argo-rollouts/issues/2707
closes: https://github.com/argoproj/argo-rollouts/issues/1914

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/) with a list of types and scopes found [here](https://github.com/argoproj/argo-rollouts/blob/master/.github/workflows/pr-title-check.yml), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

**Comment by Sineaggi** [comment]
Should fix the issue mentioned in https://github.com/argoproj/argo-rollouts/issues/1914

> When leader election is enabled there is one line that doesn't follow JSON format.
> `12:08:56.641662       1 leaderelection.go:248] attempting to acquire leader lease argocd/argo-rollouts-controller-lock...`

File: go.mod [hunk 1]
```diff
@@ -11,6 +11,7 @@ require (
 	github.com/aws/aws-sdk-go-v2/service/cloudwatch v1.25.8
 	github.com/aws/aws-sdk-go-v2/service/elasticloadbalancingv2 v1.19.8
 	github.com/blang/semver v3.5.1+incompatible
+	github.com/bombsimon/logrusr/v2 v2.0.1
```
Comment (zachaller) [inline go.mod:14]: Why v2 and not v4

Comment (Sineaggi) [reply go.mod]: Fixed

**Comment by Sineaggi** [comment]
@zachaller thanks for the review. Are there any next steps for me?

**Comment by zachaller** [comment]
@Sineaggi Merged, Thanks for contributing!

**Pull request merged**